lib/src/config.cc

```diff
@@ -304,6 +310,130 @@ namespace hocon {
         return long_list;
     }
 
+    time_duration config::get_duration(string const& path) const {
+        auto v = get_value(path);
+        if (auto d = dynamic_pointer_cast<const config_double>(v)) {
+            return milliseconds(d->long_value());
+        } else if (auto l = dynamic_pointer_cast<const config_long>(v)) {
+            return milliseconds(l->long_value());
+        } else if (auto i = dynamic_pointer_cast<const config_int>(v)) {
+            return milliseconds(i->long_value());
+        } else if (auto str = dynamic_pointer_cast<const config_string>(v)) {
+            return parse_duration(str->transform_to_string(), str->origin(), path);
+        } else {
+            throw bad_value_exception(*v->origin(), path, "Value at '" + path + "' was not a number or string.");
+        }
+    }
+
+    double config::get_duration_as_double(string const& path, time_unit unit) const {
+        auto timespan = get_duration(path);
+        int64_t nanos = timespan.total_nanoseconds();
+        switch (unit) {
+            case time_unit::NANOSECONDS:
+                return nanos;
+            case time_unit::MICROSECONDS:
+                return nanos / 1000.0;
+            case time_unit::MILLISECONDS:
+                return nanos / 1000000.0;
+            case time_unit::SECONDS:
+                return nanos / 1000000000.0;
+            case time_unit::MINUTES:
+                return nanos / 60000000000.0;
+            case time_unit::HOURS:
+                return nanos / 3600000000000.0;
+            case time_unit::DAYS:
+                return nanos / 86400000000000.0;
+            default:
+                throw config_exception("Not a valid time_unit");
+        }
+    }
+
+    int64_t config::get_duration_as_long(string const& path, time_unit unit) const {
+        auto timespan = get_duration(path);
+        int64_t nanos = timespan.total_nanoseconds();
+        switch (unit) {
+            case time_unit::NANOSECONDS:
```

Comment:
So originally I was using those, but they were behaving incorrectly. Initializing the time_duration with the max nanosecond value and then calling total_seconds on it returned a completely incorrect number. And I'm not really sure why. But calling total_nanoseconds always returns the right value, so I did the math off of that one.